src/d/actor/d_a_kita.cpp

```diff
@@ -4,52 +4,448 @@
+        cap_min_val(pActor->field_2BC.z, (REG0_F(4) + 50.0f) * std::abs(local_44.x - local_50.x));
+        float fVar1 = (REG0_F(8) + 5.0f) * std::abs(local_44.x - local_50.x);
+        if (fVar1 > 10.0f && pActor->field_2B0.x < fVar1) {
+            cLib_addCalc2(&pActor->field_2B0.x,fVar1,1.0,REG0_F(7) + 1.2f);
+        }
+        fVar1 = (REG0_F(8) + 5.0f) * std::abs(local_44.z - local_50.z);
+        if (fVar1 > 10.0f && pActor->field_2B0.z < fVar1) {
+            cLib_addCalc2(&pActor->field_2B0.z,fVar1,1.0,REG0_F(7) + 1.2f);
+        }
+        cLib_addCalc2(&pActor->mPosRel.y,REG0_F(2) + -100.0f,0.1,REG0_F(3) + 10.0f);
+    }
 }
 
 /* 00000368-00000408       .text daKita_Draw__FP10kita_class */
-static BOOL daKita_Draw(kita_class*) {
-    /* Nonmatching */
+static BOOL daKita_Draw(kita_class* i_this) {
+    dKy_getEnvlight().settingTevStruct(TEV_TYPE_BG0, &i_this->current.pos, &i_this->tevStr);
+    dKy_getEnvlight().setLightTevColorType(i_this->mModel, &i_this->tevStr);
+    dComIfGd_setListBG();
+    mDoExt_modelUpdateDL(i_this->mModel);
+    dComIfGd_setList();
+    return TRUE;
 }
 
 /* 00000408-0000126C       .text kita_move__FP10kita_class */
-void kita_move(kita_class*) {
+void kita_move(kita_class* i_this) {
     /* Nonmatching */
+    static u8 himo_off_check[4] = {1,2,4, 8};
+    static short himo_off_ya[16] = {1, 0xA000, 0x6000, 0x8000, 0xE000, 0xC000, 0xE000, 0xA000, 0x2000, 0x2000, 0x4000, 0x6000, 0, 0xE000, 0x2000, 1};
+    static short himo_off_xa[16] = {0, 0xF000, 0xF000, 0xC800, 0xF000, 0xC800, 0, 0xC000, 0xF000, 0, 0xC800, 0xC000, 0xC800, 0xC000, 0xC000, 0};
+    static short himo_off_yp[16] = {0, 0x96, 0x96, 0x190, 0x96, 0x190, 0x190,0x1F4, 0x96, 0x190, 0x190, 0x1F4, 0x190, 0x1F4, 0x1F4, 0};
+
+    short maxSpeed, ya_offset, xa_offset;
+    s32 mask;
+    daPy_py_c* player_actor = daPy_getPlayerActorClass();
+    i_this->mMoveCounter++;
+    switch(i_this->field_29A){
+        case 0:
+            mask = 0;
+            for(int i = 0; i < 4; i++){
+                if(i_this->field_318[i] == 2){
+                    i_this->field_318[i] = 0;
+                    i_this->field_320 = 0;
+                }
+                if(i_this->field_318[i] != 0){
+                    if(i_this->field_318[i] == 1){
+                        i_this->field_320 = 0;
+                        i_this->field_318[i] = 5;
+                    }
+                    mask |= himo_off_check[i];
+                }
+            }
+
+            ya_offset = himo_off_ya[mask], xa_offset = himo_off_xa[mask];
+            cLib_addCalc2(&i_this->mHeight, REG0_F(4) + (-static_cast<float>(himo_off_yp[mask]) * 0.3f), 0.05, i_this->field_320 * 250.0f);
+            if(mask == 0b1111){
+                i_this->field_29A = 1;
+                dBgS_GndChk solid_ground_check;
+                /* Nonmatching */ solid_ground_check.m_pos.set(i_this->current.pos.x, i_this->current.pos.y - 200.0f, i_this->current.pos.z);
```

Comment:
The reason it doesn't match may be because the debug map shows they used the `SetPos(Vec* pos)` inline here, while you're assigning to it directly. Try this:
```suggestion
                Vec pos = {i_this->current.pos.x, i_this->current.pos.y - 200.0f, i_this->current.pos.z};
                solid_ground_check.SetPos(&pos);
```